fields/types/gcsimage/GcsImageField.js

```diff
@@ -1 +1,203 @@
-module.exports = require('../cloudinaryimage/CloudinaryImageField');
+import Field from '../Field';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import classnames from 'classnames';
+import get from 'lodash/get';
+import { Button, FormField, FormInput, FormNote } from 'elemental';
+
+const _ = {
+  get,
+};
+
+const SUPPORTED_TYPES = ['image/gif', 'image/png', 'image/jpeg', 'image/bmp', 'image/x-icon', 'application/pdf', 'image/x-tiff', 'image/x-tiff', 'application/postscript', 'image/vnd.adobe.photoshop', 'image/svg+xml'];
+
+module.exports = Field.create({
+
+  displayName: 'GcsImageField',
+
+  getThumbnail () {
+    return _.get(this.props, 'value.resizedTargets.mobile.url', '');
+  },
+
+  getLocation () {
+    const gcsBucket = _.get(this.props, 'value.gcsBucket', '');
+    const gcsDir = _.get(this.props, 'value.gcsDir', '');
+    const filename = _.get(this.props, 'value.filename', '');
+
+    if (gcsBucket && gcsDir && filename) {
+      if (typeof gcsDir === 'string' && gcsDir.slice(-1) === '/') {
+        return `gs://${gcsBucket}/${gcsDir}${filename}`;
+      }
+      return `gs://${gcsBucket}/${gcsDir}/${filename}`;
+    } else if (gcsBucket && filename) {
+      return `gs://${gcsBucket}/${filename}`;
+    }
+
+    return '';
+  },
+
+  /**
+   *  is this image uploaded from end user local workspace in this session?
+   */
+  hasLocal () {
+    return this.state.origin === 'local';
+  },
+
+  hasImage () {
+    return this.getLocation() || this.hasLocal();
+  },
+
+  getImageSource () {
+    if (this.hasLocal()) {
+      return this.state.localSource;
+    } else if (this.getThumbnail()) {
+      return this.getThumbnail();
+    } else {
+      return null;
+    }
+  },
+
+  /**
+   * Render an image preview
+   */
+  renderImagePreview () {
+    const thumbnail = this.getImageSource();
+
+    if (thumbnail) {
+      return (
+        <div className="image-preview">
+          <a className="img-thumbnail" href={thumbnail} target="_blank">
+            <img className="img-load" style={{ height: '90' }} src={thumbnail} alt="圖片無法載入，請稍後再試。"/>
+          </a>
+        </div>
+      );
+    }
+
+    return null;
+  },
+
+  /**
+   * Render image details
+   */
+  renderImageDetails () {
+    if (this.getLocation()) {
+      const iptc = _.get(this.props, 'value.iptc', {});
+      let createdTime = _.get(iptc, 'created_time', '')
```

Comment:
Do we need to handle the timezone issue here?

Reply:
I need to check timezone with photographer.
As far as I know, we only have created date and created time.

Reply:
I suppose both of the `created date` and `time` are from system timezone generated from photo editing tools: `Lightroom` or `Photoshops`. My concern is whether the `time data` is handled correctly during the parsing from `UTC` timezone.

Reply:
The program does not parse `created time` and `created date` from `UTC` timezone.
It just shows the text values.